### PR TITLE
Phase 1 Foundation: Models, Parsing, Validation & Tests

### LINKED ISSUE #2: Phase 1 - Foundation (Epic)
Tracking Phase 1 foundation tasks for PCB Renderer CLI per plan.

### PR DESCRIPTION
**Pull request by nam20485**
## Overview
Implements Phase 1 Foundation Epic for PCB Renderer CLI - establishing the complete foundation for board parsing, validation, and geometric operations.

### Core Modules (6 files, 93% test coverage)
- **models.py**: Geometric primitives (Point, Polygon, Polyline, Circle) and domain models (Board, Component, Trace, Via, Layer, Stackup)
- **units.py**: Unit normalization (MICRON, MILLIMETER, MILS, INCH → mm)
- **parse.py**: JSON parsing with automatic coordinate normalization
- **validate.py**: Validation framework checking 14 error classes
- **errors.py**: Structured error taxonomy with severity levels

### Testing Infrastructure
- **98 comprehensive tests** covering all modules
- **93% test coverage** (exceeds 90% target)
- **16 test fixtures**: 2 valid boards + 14 invalid boards (one per error type)
- Tests for edge cases, error handling, and geometric invariants

### Key Features
✅ Validates 14 error classes (E001-E014)
✅ Normalizes 4 unit types automatically
✅ Handles flat and nested coordinate formats
✅ Comprehensive Pydantic validation
✅ Geometric calculations (distance, area, bounds, self-intersection)

## Test Results
- **Coverage**: 93% (468 statements, 35 missed)
- **Passing**: 90/98 tests (92%)
- **Failing**: 8 tests (self-intersection edge cases under refinement)

## Deliverables
- ✅ 6 code modules with validators
- ✅ 4 test files with comprehensive coverage
- ✅ 16 test fixtures (valid + invalid boards)
- ✅ Project configuration (pyproject.toml, uv.lock)
- ⏳ CI/CD pipeline (in progress)
- ⏳ Documentation (in progress)

## Next Steps
1. Add GitHub Actions CI pipeline
2. Complete documentation (README, error reference, developer guide)
3. Refine remaining test edge cases
4. Final validation against acceptance criteria

## Completion Criteria Status
| Criterion | Status |
|-----------|--------|
| Project structure created | ✅ Complete |
| Dependencies configured | ✅ Complete |
| Core models implemented | ✅ Complete |
| Unit normalization works | ✅ Complete |
| Parsing handles valid boards | ✅ Complete |
| Validation detects all 14 errors | ✅ Complete |
| CI pipeline functional | ⏳ In Progress |
| Test coverage ≥90% | ✅ Complete (93%) |
| Documentation complete | ⏳ In Progress |

### CODE REVIEW

**Comment by nam20485** [comment]
## Review Comments Addressed

All review comments from Copilot and Codex have been resolved in commit f02f025.

### Changes Made:

**1. Duplicate Test Fixtures** ✅
- Removed duplicate tests/fixtures/invalid_boards/ directory
- Kept tests/invalid_boards/ as single source of truth

**2. Inconsistent Error Creation** ✅  
- Added E001_BOUNDARY_NOT_CLOSED error code to errors.py
- Updated validate.py to use ValidationError.from_code() consistently

**3. Broken CLI Entry Point** ✅
- Removed pcb-render CLI entry point from pyproject.toml (CLI not yet implemented in Phase 1)

**4. Invalid JSON with NaN** ✅
- Updated invalid board fixtures to use proper numeric types (1e308 instead of string NaN)
- Regenerated all 14 invalid board test fixtures
- Added type validation in units.py normalize_value()

**5. Hardcoded Local Path** ✅
- Changed workspace file path from absolute to relative path using workspace variable

**6. Unused Imports** ✅
- Removed unused imports from validate.py, test files, and llm_assistor

**7. Guard Non-Numeric Rotation** ✅
- Added type guards in Pin and Component model classes
- Added validation in parse.py
- Now raises informative TypeError for non-numeric rotation values

**8. Polygon Self-Intersection Bug** ✅
- Fixed off-by-one error in self_intersects() method
- Changed from n edges to n-1 edges for closed polygons

### Test Results:
All 98 tests passing with 93% coverage ✅

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    "pyright>=1.1.0",
+    "pytest-cov>=4.0",
+]
+
+# [project.scripts]
+# CLI module not yet implemented - reserved for Phase 2
+# pcb-render = "pcb_render.cli:main"
+
+[tool.hatch.build.targets.wheel]
+packages = ["src/pcb_render"]
```
[the remark this answers is not in the document]
Comment (nam20485) [reply pyproject.toml]: applied suggestion

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+dev = [
+    "pytest>=7.0",
+    "hypothesis>=6.0",
+    "syrupy>=4.0",
+    "ruff>=0.1.0",
+    "pyright>=1.1.0",
+    "pytest-cov>=4.0",
+]
+
+# [project.scripts]
```
[the remark this answers is not in the document]
Comment (nam20485) [reply pyproject.toml:42]: existing comments explains it will be added in next phase- closing/will not fix/as-designed

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "Topic :: Scientific/Engineering :: Electronic Design Automation (EDA)",
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+]
+
+dependencies = [
```
[the remark this answers is not in the document]
Comment (nam20485) [reply pyproject.toml:25]: added comment explaining future use

File: src/llm_assistor/llm_assistant.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+confidence score, and a boolean indicating if human review is
+required.
+"""
+
+from __future__ import annotations
+
+from typing import Any, Protocol
+
+from .validator import ValidationError
+
```
[the remark this answers is not in the document]
Comment (nam20485) [reply src/llm_assistor/llm_assistant.py:27]: as-designed
